src/nvim/ex_docmd.c

```diff
@@ -9411,3 +9411,40 @@ static void ex_terminal(exarg_T *eap)
     xfree(name);
   }
 }
+
+static void ex_plug(exarg_T *eap)
+{
+  char *path = NULL;
+  if (*eap->arg != NUL) {
+    if (os_isdir(eap->arg)) {  // full path: plug ~/.config/nvim/plugged/coolplugin
+      path = (char *)vim_strsave_fnameescape(eap->arg, FALSE);
+    } else {  // spec: plug neovim/coolplugin
+      char_u *basename = path_tail((char_u *)eap->arg); // 'neovim/coolplugin' -> 'coolplugin'
+      if (STRNCMP(basename, "", 1) != 0 && STRNCMP(p_plugindir, "", 1) != 0) {
+        if (!os_isdir(p_plugindir)) {
+          // we build a path using the first item in 'runtimepath' (usually, '~/.config/nvim')
+	  char_u *rtp_copy = vim_strsave(p_rtp);
+	  char_u *user_vim = xmallocz(MAXPATHL);
```

Comment:
You don’t need to allocate `user_vim`, just use `NameBuff`. This only made any sense if `path` survived the function call.